### ISSUE TITLE
Implement an upper limit on the number of Realm files that can be opened at once

### ISSUE THREAD

**Issue by TimOliver** [realm/realm-browser-osx#41]
In instances such as the Realm Plugin where multiple Realm files can be programmatically opened at once in the Browser, we'll need to implement an upper limit to ensure that the Browser does't get overloaded.

At present, if a sufficiently large number of Realm files is passed to the Browser, it'll attempt to open them all until it runs out of resources.

**Comment by bmunkholm** [comment]
How would you define that limit? What happens when it is overloaded?

**Comment by TimOliver** [comment]
At present this is what happens if it gets overloaded:
![6d5e0380-24c8-11e5-8272-ed656ddbee34](https://cloud.githubusercontent.com/assets/429119/8591006/04fef5c4-2658-11e5-876a-aac6d4ba16fa.jpg)

For the time being, we're going to add a prompt to the Browser for the user to confirm if the plugin tries to open more than 5 .realm files at once.

**Comment by jpsim** [comment]
@TimOliver this should be a very simple fix, could you dedicate a few minutes when you're back online to address this?

**Comment by bmunkholm** [comment]
It this a real problem? Who wants to open that many files? And if they do,
there is an error informing about what's wrong.

**Comment by TimOliver** [comment]
@jpsim No worries. I've added a new method that prompts the user if the delegate detects multiple files being opened at once. I've tested it by selected multiple Realm files on disk and dragging them over the Browser's icon.

 Unfortunately, I haven't been able to test it against the Realm Plugin properly yet. That scenario where having two versions of Xcode on the system confuses the plugin happened to me as of today (i.e., I have Xcode-beta.app installed for running apps on my iOS 9 devices and now Realm Plugin cannot see any Realm apps on my computer), so I'm going to need to create some specialized tests on the Browser's behalf to simulate the Realm Plugin's behavior until that one's sorted out.

@bmunkholm Yep! It's a real problem. That screenshot of mine is what happened when I was testing out the Realm Plugin against my own app's usage of Realm. My app creates a discrete Realm file for each user document for the purposes of storing cached information (Upwards of 500 Realm files in production!), so when that was tried against Realm Plugin, it tried to open every single one automatically, and the Browser couldn't cope.

Thankfully, the number of Realm files that need to be open before the Browser complains is very high (Probably more than 100 Realm files at once), so it's not something you would see under most circumstances. I've now set it that a warning appears if you try and open more than 20 files in one operation.
